compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -2173,17 +2174,37 @@ public void visit(BLangIf ifNode) {
     @Override
     public void visit(BLangQueryAction queryAction) {
         SymbolEnv prevQueryEnv = this.queryEnv;
+        boolean prevWithinQuery = withinQuery;
+        if (isInQuery(queryAction.parent)) {
+            this.withinQuery = true;
+        }
         queryAction.getQueryClauses().forEach(this::acceptNode);
+        this.withinQuery = prevWithinQuery;
         this.queryEnv = prevQueryEnv;
     }
 
     @Override
     public void visit(BLangQueryExpr queryExpr) {
         SymbolEnv prevQueryEnv = this.queryEnv;
+        boolean prevWithinQuery = withinQuery;
+        if (isInQuery(queryExpr.parent)) {
+            this.withinQuery = true;
+        }
         queryExpr.getQueryClauses().forEach(this::acceptNode);
+        this.withinQuery = prevWithinQuery;
         this.queryEnv = prevQueryEnv;
     }
 
+    private boolean isInQuery(BLangNode parent) {
+        while (parent != null) {
+            if (isQueryExprOrAction(parent)) {
+                return true;
+            }
+            parent = parent.parent;
+        }
+        return false;
```

Comment:
Do we have tests to cover this line?

Reply:
Modified the code since this line is never reached.